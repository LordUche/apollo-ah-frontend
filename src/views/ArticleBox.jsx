import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import compareDateFormat from 'date-fns/distance_in_words_strict';

export default function ArticleBox({ article }) {
  return (
    <div className="col-sm-6 col-lg-4">
      <div className="article">
        <Link to={`/article/${article.slug}`} className="link-surface" />
        <div className="article-cover">
          <img src={article.image} className="article-cover-img" alt={article.title} />
        </div>
        <div className="article-item">
          <span className="article-category">{article.articleCategory.category}</span>
          <h4>{article.title}</h4>
          <p>{article.description.substring(0, 100)}</p>
          <div className="article-author">
            <img src={article.User.Profile.image} alt={article.User.Profile.username} />
            <div>
              <h6>{article.User.Profile.firstname || article.User.Profile.username}</h6>
              <span>
                {compareDateFormat(article.createdAt, new Date())}
                {' '}
                 ago
              </span>
              <span className="article-author-dot" />
              <span>{article.readTime}</span>
            </div>
          </div>
        </div>
      </div>
    </div>

  );
}
ArticleBox.propTypes = {
  article: PropTypes.object.isRequired,
};
