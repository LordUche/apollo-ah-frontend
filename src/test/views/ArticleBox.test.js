import React from 'react';
import { shallow } from 'enzyme';
import ArticleBox from '../../views/ArticleBox';


describe('<ArticleBox>', () => {
  it('should render without crashing', () => {
    const wrapper = shallow(<ArticleBox article={{
      slug: '',
      articleCategory: {},
      description: '',
      User: {
        Profile: {}
      },
    }} />);
    expect(wrapper).toMatchSnapshot();
  });
});
