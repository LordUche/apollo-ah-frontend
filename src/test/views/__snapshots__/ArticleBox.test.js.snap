// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ArticleBox> should render without crashing 1`] = `
<div
  className="col-sm-6 col-lg-4"
>
  <div
    className="article"
  >
    <Link
      className="link-surface"
      replace={false}
      to="/article/"
    />
    <div
      className="article-cover"
    >
      <img
        className="article-cover-img"
      />
    </div>
    <div
      className="article-item"
    >
      <span
        className="article-category"
      />
      <h4 />
      <p />
      <div
        className="article-author"
      >
        <img />
        <div>
          <h6 />
          <span>
            NaN years
             
            ago
          </span>
          <span
            className="article-author-dot"
          />
          <span />
        </div>
      </div>
    </div>
  </div>
</div>
`;
